src/SFML/Graphics/Font.cpp

```diff
@@ -192,32 +189,31 @@ bool Font::loadFromFile(const std::filesystem::path& filename)
     {
         err() << "Failed to load font (failed to set the Unicode character set)\n"
               << formatDebugPathInfo(filename) << std::endl;
-        return false;
+        return std::nullopt;
     }
 
     // Store the loaded font handles
-    m_fontHandles = std::move(fontHandles);
+    result.m_fontHandles = std::move(fontHandles);
 
     // Store the font information
-    m_info.family = face->family_name ? face->family_name : std::string();
+    result.m_info.family = face->family_name ? face->family_name : std::string();
 
-    return true;
+    return {result};
 
 #else
 
-    m_stream = std::make_unique<priv::ResourceStream>(filename);
-    return loadFromStream(*m_stream);
+    Font result;
+    result.m_stream = std::make_unique<priv::ResourceStream>(filename);
+    return loadFromStream(*result.m_stream);
```

Comment:
I think this will not work. There is an ownership problem here. `result` is a temporary owning the stream that will be deleted when the function is exited.
Maybe something like this would be better (untested):
```c++
    auto stream = std::make_unique<priv::ResourceStream>(filename);
    if(auto result = loadFromStream(*stream))
    {
        result->m_stream = std::move(stream);
        return result;
    }
    return std::nullopt;
```